src/output/tts/tts_online.py

```diff
@@ -19,6 +19,6 @@ def speak(self, text):
         sound.seek(0)
         pygame.mixer.music.load(sound, "mp3")
         pygame.mixer.music.play()
-        while pygame.mixer.music.get_busy():
-            pygame.time.delay(10)
-            pygame.event.poll()
+        for event in pygame.event.get():
```

Comment:
is this used to allow closing the program while the TTS is talking (because we're freezing the main thread for this period as we're not doing TTS in a separate thread)?

Reply:
yes, after closing the window, the speech is stil running for the last pattern

Reply:
yes, after closing the window, the speech for the last pattern ran.

Reply:
yes, after closing the window, the speech for the last pattern was still running to the end